content/components/tooltip.mdx

```diff
@@ -3,42 +3,239 @@ title: Tooltip
+    />
+    <Caption>Use a tooltip to give a text label to an icon button</Caption>
+  </Do>
+  <Dont>
+    <img
+      width="456"
+      alt="Non-interactive text element activating a long and wordy tooltip on hover"
+      src="https://github.com/primer/primitives/assets/18661030/1d8d86a2-8ee7-40d2-8934-c3d36f34d9b3"
+    />
+    <Caption>Don't use tooltips on non-interactive elements</Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img
+      width="800"
+      alt=""
+      src="https://github.com/primer/primitives/assets/18661030/d5528bb1-cf2e-41f6-b029-d861fb28375b"
+    />
+    <Caption>Do use tooltips for icon buttons that serve unique purposes</Caption>
+  </Do>
+  <Dont>
+    <img
+      width="800"
+      alt=""
+      src="https://github.com/primer/primitives/assets/18661030/9713ad8b-9f52-44fd-8b46-06d5151b813f"
+    />
+    <Caption>
+      Don’t use tooltips for icon buttons that appear in a group of other controls which share the same functionality,
+      such as opening a context menu.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+## Options
+
+### Label
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="icon button with tooltip as a label"
+    src="https://github.com/primer/design/assets/18661030/b06d4fc2-d4e6-482d-a9b9-05eb866ce3e1"
+  />
+  <Box as="p" mt="0">
+    For interactive controls with no visible text label, a tooltip can be used to provide one. This label should be
+    short and descriptive just as you would for a button with visible text.
+  </Box>
+</Box>
+
+### Description
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="icon button with tooltip as a description"
+    src="https://github.com/primer/design/assets/18661030/2d0cd085-2454-406e-8c99-bc8bd3dd20db"
+  />
+  <Box as="p" mt="0">
+    For interactive controls that require additional context, a tooltip can be used to provide it. The `label` is also
+    required, and the description will be announced to a screen reader in addition to the label. Keep the description
+    short and descriptive.
+  </Box>
+</Box>
+
+### Position
+
+By default, the Tooltip component will position itself below the interactive control in a centered alignment. The position is dependant on the space available within the viewport, and it will never break outside of it. It may appear to the top, bottom, left, or right of the interactive control.
+
+<img
+  width="960"
+  role="presentation"
+  src="https://github.com/primer/design/assets/18661030/c620d282-1efd-4556-9d9c-770c37359fc9"
+  alt="tooltips rendered in all possible positions surrounding a control"
+/>
+
+## Alternatives to tooltips
```

Comment:
We have a couple of lint rules and doc pages where we cross-reference the existing [Tooltip alternatives](https://primer.style/design/guides/accessibility/tooltip-alternatives) docs. Let's make sure to update those to avoid a dead link! (I can verify tomorrow which ones those are). 

Or could it be sufficient to just cross-reference that page from this section rather than moving it entirely?

Reply:
* [Primer CSS Tooltipped](https://primer.style/design/guides/development/rails/migration-guides/primer-css-tooltipped)
* [a11y-role-supports-aria-props](https://github.com/github/eslint-plugin-github/blob/b93f04ae4600c18f86d65e4d5854820f71ed26bb/docs/rules/a11y-role-supports-aria-props.md?plain=1#L16)
* [NoAriaLabelMisuse](https://github.com/github/erblint-github/blob/main/docs/rules/accessibility/no-aria-label-misuse.md?plain=1#L10)
* [a11y-no-title-attribute](https://github.com/github/eslint-plugin-github/blob/b93f04ae4600c18f86d65e4d5854820f71ed26bb/docs/rules/a11y-no-title-attribute.md?plain=1#L11)
* [a11y_no_rogue_role_equals_tooltip](https://github.com/github/github/blob/5dc32d30d7d08078c30df315fd2b6f750abc6ba8/.erb-linters/accessibility/a11y_no_rogue_role_equals_tooltip.rb#L15)
* [PVC Tooltip](https://github.com/primer/view_components/blob/main/app/components/primer/alpha/tooltip.rb#L6)

Reply:
I'll follow up with the links outside this repo once this PR merges! ❤️ thank you